Calculator/Calculator/Model/CalculatorItemQueue.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  CalculatorItemQueue.swift
+//  Calculator
+//
+//  Created by junho lee on 2022/09/20.
+//
+
+class CalculatorItemQueue<T: CalculateItem> {
+    class LinkedList<T> {
+        class Node<T> {
+            var value: T
+            var next: Node?
+            
+            init(value: T, next: Node? = nil) {
+                self.value = value
+                self.next = next
+            }
+        }
+        
+        private var head: Node<T>?
+        private var tail: Node<T>?
+        
+        func addLast(_ element: T) {
+            guard let last = tail else {
+                tail = Node(value: element)
+                head = tail
+                return
+            }
+            let newNode = Node(value: element)
+            last.next = newNode
+            tail = newNode
+        }
```

Comment:
tail을 옵셔널바인딩하는 이유가 nil인지 아닌지를 체크하기 위함인가요? 이후 코드에는 굳이 옵셔널 바인딩을 한 이유를 찾을 수가 없네요. 또한 이 guard문이 어떤 역할을 하는지를 알려면 코드를 해석해야합니다.

아래와 같이 조건에 이름을 지어주는건 어떨까요?

```suggestion
        func addLast(_ element: T) {
            if isEmpty {
                tail = Node(value: element)
                head = tail
            } else {
                let newNode = Node(value: element)
                last.next = newNode
                tail = newNode
            }
        }
        
        private var isEmpty {
            return tail == nil
        }
```

Reply:
* 2bd3e16에서 LinkedList를 제거했습니다.